plugins/woocommerce-admin/client/marketing/data-multichannel/selectors.ts

```diff
@@ -13,9 +13,14 @@ export const getRecommendedChannels = ( state: State ) => {
 
 /**
  * Get campaigns from state.
+ *
+ * @param state   State passed in from the data store.
+ * @param page    Page number. First page is `1`.
+ * @param perPage Page size, i.e. number of records in one page.
  */
-export const getCampaigns = ( state: State ) => {
-	return state.campaigns;
+export const getCampaigns = ( state: State, page: number, perPage: number ) => {
+	const key = `${ page }-${ perPage }`;
```

Comment:
❔ Related to the other comment about using `-` as data separator in key: it might be good to refactor this key construction into a shared function that we can use in selector and reducer.

Reply:
Replied in https://github.com/woocommerce/woocommerce/pull/41211#discussion_r1411813274.